### PR TITLE
Migrate to new android gradle plugin

### PR DESCRIPTION
**Pull request by NikolayManzhos**

### CODE REVIEW

**Comment by NikolayManzhos** [comment]
CI failing due to ObsoleteLintCustomCheck, there are 3 custom checks which lead to this issue
```butterknife.lint.InvalidR2UsageDetector```, ```nl.littlerobots.rxlint.SubscribeDetector,``` ```timber.lint.WrongTimberUsageDetector```
For ButterKnife and Timber there is a fix availble in latest SNAPSHOT.
Sadly i cannot find any fixed version for RxLint.
So i guess the only way for now is to disable ```ObsoleteLintCustomCheck```.
What do you think?

**Comment by vanniktech** [comment]
RxLint 1.6 should work with 3.0.0 if I recall correctly

File: build.gradle [hunk 1]
```diff
@@ -50,6 +48,7 @@ buildscript {
 allprojects {
     repositories {
         jcenter()
+        maven { url "https://oss.sonatype.org/content/repositories/snapshots/"}
```
Comment (artem-zinnatullin) [inline build.gradle]: We definitely can't depend on snapshots, that can break any future build…

File: build.gradle [hunk 2]
```diff
@@ -8,16 +8,14 @@ buildscript {
...
     dependencies {
         // See dependencies.gradle for declaration of plugins.
 
         classpath gradlePlugins.android
 
-        // Backport of some Java 8 language features such as Lambdas and method references.
-        classpath gradlePlugins.retrolambda
-
         // Needed for Lint + Retrolambda.
         classpath gradlePlugins.lombok
```
Comment (artem-zinnatullin) [inline build.gradle]: This can be removed as well

File: build.gradle (hunk 1 above)
Comment (NikolayManzhos) [reply build.gradle]: Alright, so should we just ignore ```ObsoleteLintCustomCheck``` until new versions of ButterKnife and Timber is released?

Comment (artem-zinnatullin) [reply build.gradle]: I'm not a fan of JitPack, but that could be an option here, or just wait couple days for Jake to properly release new versions :)

Comment (vanniktech) [reply build.gradle]: Maybe if you create an issue and ask nicely we'll get those published faster.

Comment (artem-zinnatullin) [reply build.gradle]: Looks like ButterKnife 9.x is far from release https://github.com/JakeWharton/butterknife/milestone/5

**Comment by artem-zinnatullin** [comment]
Looks great! Thank you!

**Pull request merged**